rust/src/push/evaluator.rs

```diff
@@ -256,9 +256,11 @@ impl PushRuleEvaluator {
         };
 
         let result = match known_condition {
-            KnownCondition::EventMatch(event_match) => {
-                self.match_event_match(&event_match.key.clone(), &event_match.pattern.clone())?
-            }
+            KnownCondition::EventMatch(event_match) => self.match_event_match(
+                &self.flattened_keys,
```

Comment:
In both cases you pass `&self.flattened_keys` to self.match_event_match. Is that first argument to `match_event_match` maybe redundant?

Reply:
We also call `match_event_match` from `match_related_event_match` where we pass in a different value.